File: FrontEnd/stock-analyzer/src/app/types/Filter.ts
import { Company } from "./Company"

export type Filter = {
    filterName: string,
    series: string,
    filterType: string,
    criterias: FilterCriteria[]
}

export type FilterCriteria = {
    sequence: number,
    fieldName: string,
    changeType: string,
    logicalOperator: string,
    periodType: string,
    periodValue: number
}
// Symbol, series, date, Avg fieldName(periodValue'D/W/M/Y')^
export type FilterResult = {
    id: number,
    filterCriteria: FilterCriteria,
    calculationDate: Date,
    company: Company,
    value: number
}

export enum ChangeType {
    Increase = 'Increase',
    Decrease = 'Decrease'
}

export enum LogicalOperator {
    And = 'And',
    Or = 'Or'
}

export enum PeriodType {
    Days = 'Days',
    Weeks = 'Weeks',
    Months = 'Months',
    Years = 'Years',
}

export enum filterType {
    MovingAverage = "MovingAverage",
    Continuous = "Continuous"
}

File: Stock Analyzer/Stock Analyzer Domain/Models/Filter/Filter.cs

namespace Stock_Analyzer_Domain.Models.Filter
{
  public class Filter
  {
    public Filter(string filterName, string series, FilterType filterType)
    {
      FilterName = filterName;
      Series = series;
      FilterType = filterType;
      Criterias = new List<FilterCriteria>();
    }

    public Guid Id { get; set; }

    public string FilterName { get; set; }

    public string Series { get; set; }

    public FilterType FilterType { get; set; }

    public List<FilterCriteria> Criterias { get; set; } = new List<FilterCriteria>();
  }
}
